pex/resolver.py

```diff
@@ -205,57 +207,64 @@ def resolve(requirements=None,
       to_build.extend(local_projects_from_requirement_file(requirement_file))
 
-  to_copy = []
+  to_install = []
   if os.path.exists(resolved_dists):
     for distribution in os.listdir(resolved_dists):
       path = os.path.join(resolved_dists, distribution)
       if os.path.isfile(path) and path.endswith('.whl'):
-        to_copy.append(path)
+        to_install.append(path)
       else:
         to_build.append(path)
 
-  if not any((to_build, to_copy)):
+  if not any((to_build, to_install)):
     # Nothing to build or install.
     return []
 
-  safe_mkdir(built_wheels)
-
   if to_build:
-    try:
-      build_wheels(distributions=to_build,
-                   target=built_wheels,
-                   cache=cache,
-                   interpreter=interpreter)
-    except PipError as e:
-      raise Untranslateable('Failed to build at least one of {}:\n\t{}'.format(to_build, str(e)))
-
-  if to_copy:
-    for wheel in to_copy:
-      dest = os.path.join(built_wheels, os.path.basename(wheel))
-      TRACER.log('Copying downloaded wheel from {} to {}'.format(wheel, dest))
-      shutil.copy(wheel, dest)
+    with TRACER.timed('Building distributions:\n  {}'.format('\n  '.join(to_build))):
+      try:
+        build_wheels(distributions=to_build,
+                     target=built_wheels,
+                     cache=cache,
+                     interpreter=interpreter)
+      except PipError as e:
+        raise Untranslateable('Failed to build at least one of {}:\n  {}'
+                              .format(', '.join(to_build), str(e)))
+      to_install.extend(os.path.join(built_wheels, wheel) for wheel in os.listdir(built_wheels))
 
   # 3. Chroot
   resolved_distributions = []
 
-  for wheel_file in os.listdir(built_wheels):
-    chroot = os.path.join(installed_chroots, wheel_file)
-    try:
-      install_wheel(wheel=os.path.join(built_wheels, wheel_file),
-                    target=chroot,
-                    compile=compile,
-                    overwrite=True,
-                    cache=cache,
-                    interpreter=interpreter)
-    except PipError as e:
-      raise Untranslateable('Failed to install {}:\n\t{}'.format(wheel_file, str(e)))
-
-    environment = Environment(search_path=[chroot])
-    for dist_project_name in environment:
-      resolved_distributions.extend(environment[dist_project_name])
-
-  markers_by_req_key = _calculate_dependency_markers(resolved_distributions,
-                                                     interpreter=interpreter)
+  with TRACER.timed('Installing distributions:\n  {}'.format('\n  '.join(to_install))):
+    for wheel_file_path in to_install:
+      wheel_file = os.path.basename(wheel_file_path)
+      chroot = os.path.join(installed_chroots, wheel_file)
+      if os.path.exists(chroot):
+        TRACER.log('Using cached installation of {} at {}'.format(wheel_file, chroot))
+      else:
+        TRACER.log('Installing {} in {}'.format(wheel_file_path, chroot))
```

Comment:
Nit: `in` makes it sound like you are installing it in the chroot directory. Really, the `chroot` variable here is the specific file location, not the parent directory. A simple fix would be to change the proposition from `in` to `at`.

Better, I think, is to rename `chroot` to something like `chrooted_wheel_path` or something that more clearly indicates this is the chrooted file location, not the general chrooted directory location.

Reply:
> Nit: `in` makes it sound like you are installing it in the chroot directory.

I am actually. The chroot directory is like a mini-`site-packages` that will contain exactly one distribution after this operation, the wheel file we install there (and which is exploded + other things by pip in the process). To be super clear hopefully:

```
$ rm -rf ~/.pex
$ python -mpex requests -- -c ''
$ tree -d -L 2 ~/.pex/build/chroots/
/home/jsirois/.pex/build/chroots/
├── certifi-2019.9.11-py2.py3-none-any.whl
│   ├── certifi
│   └── certifi-2019.9.11.dist-info
├── chardet-3.0.4-py2.py3-none-any.whl
│   ├── bin
│   ├── chardet
│   └── chardet-3.0.4.dist-info
├── idna-2.8-py2.py3-none-any.whl
│   ├── idna
│   └── idna-2.8.dist-info
├── requests-2.22.0-py2.py3-none-any.whl
│   ├── requests
│   └── requests-2.22.0.dist-info
└── urllib3-1.25.7-py2.py3-none-any.whl
    ├── urllib3
    └── urllib3-1.25.7.dist-info

16 directories
```

In that example, taking one `chroot` of `~/.pex/build/chroots/requests-2.22.0-py2.py3-none-any.whl`, you'll see that installed in it is exactly one distribution, `requests-2.22.0`.

Reply:
Perhaps it's confusing that the chroot dir basename is the zipped wheel basename? That is a hack that allows PEXEnvironment to activate the correct set of wheels at runtime as explained in the comment up above.

Reply:
What confused me is thinking of `chroot` as one singular thing and not realizing you are referring to each wheel as having its own chroot. I was conceptualizing `chroot` to be the parent folder `/home/jsirois/.pex/build/chroots/`. 

Renaming `chroot` to `wheel_chroot` would have helped to figure that out.